Elf tut update (#120)
castep-docs/castep-docs: docs/tutorials/Bonding_and_Charge/ELF.md

A lot of the steps on examining the data are the same as in [Plotting Charge Density](charge_density.md) - if you prefer, you may use that (and its accompanying [video tutorial](https://www.youtube.com/watch?v=_c2Hk4jxmm4&feature=youtu.be)), though the cells and planes used are slightly different. 

## NaCl

We will start by looking at how electrons in an NaCl crystal behave - to illustrate what ELF exactly does and doesn't do, we will compare it to the results of a standard population analysis.

For both calculations we will use the `cell` file

*NaCl.cell*
```
%block lattice_cart
5.693 0.000 0.000
0.000 5.693 0.000
0.000 0.000 5.693
%endblock lattice_cart

%block positions_frac
Na 0.0 0.0 0.0
Na 0.0 0.5 0.5
Na 0.5 0.0 0.5
Na 0.5 0.5 0.0
Cl 0.5 0.5 0.5
Cl 0.5 0.0 0.0
Cl 0.0 0.5 0.0
Cl 0.0 0.0 0.5
%endblock positions_frac
```
and the `param` file

*NaCl.param*
```
xc_functional : LDA
cutoff_energy : 500 eV
spin_polarised : false
WRITE_FORMATTED_DENSITY : TRUE
CALCULATE_ELF : TRUE
WRITE_FORMATTED_ELF : TRUE
```

> Running castep will yield 2 files of interest: `NaCl.elf_fmt` and `NaCl.den_fmt`. We will want to compare the 2 outputs, so let's make 2 copies of the `cell` file by copying `NaCl.cell` as `NaCl_elf.cell` and `NaCl_den.cell`, and copying `NaCl.elf_fmt` to `NaCl_elf.charg_frm` and `NaCl.den_fmt` to `NaCl_den.charg_frm`. In both `.charg_frm` files remove the header, so they both start similarly to

Reviewer: This might be better as an enumeration of steps:
```suggestion
Running castep will yield 2 files of interest: `NaCl.elf_fmt` and `NaCl.den_fmt`. 
We will want to compare the 2 outputs, so:
1. Make 2 copies of the `cell` file by copying `NaCl.cell` to `NaCl_elf.cell` and `NaCl_den.cell`. 2. Copy `NaCl.elf_fmt` to `NaCl_elf.charg_frm`
3. Copy `NaCl.den_fmt` to `NaCl_den.charg_frm`. 
4. In both `.charg_frm` files remove the header.

Following this, they should look like:
```

```
1     1     1            0.000000
2     1     1            0.000000
3     1     1            0.000000
4     1     1            0.000000
5     1     1            0.000000
```

> Now download the 4 new files.

Reviewer: Download?

!!! note

>     If you wish to move the files around, make sure that the `.charg_frm` file is located in the same directory as its respective `cell` file - so `NaCl_elf.charg_frm` should be next to `NaCl_elf.cell` for example

Reviewer: ```suggestion
    If you wish to move the files around, make sure that the `.charg_frm` file is located in the same directory as its respective `cell` file - for example `NaCl_elf.charg_frm` should be with `NaCl_elf.cell`.
```

> We will first look at the density output. Open it in Vesta, and, at the top (same row as where the `Files` tab is), go to `Utilities -> 2D Data Display`. It should open a window looking like

Reviewer: Link to VESTA or the software docs page?

Reviewer: This is usually called the "menu-bar" or "top-bar"

Reviewer: ```suggestion
We will first look at the density output. Open it in Vesta, and, at the top (same row as where the `Files` tab is), go to `Utilities -> 2D Data Display`. 

It should open a window which looks like:
```

![Data display window](data_display_window.png){width="40%")}

There, click `Slice` - this will open another window

![Slice window](slice_window.png){width="40%"}

> You should keep the `hkl` (lattice Miller indices) as `001`, but make a 2x2 supercell but changing `x(min)`, `y(min)` and `z(min)` to `-1`. Click `Ok` and you should get an output looking like

Reviewer: ```suggestion
You should keep the `hkl` (lattice Miller indices) as `001`, but make a 2x2 supercell and change `x(min)`, `y(min)` and `z(min)` to `-1`. 

Click `Ok` and you should get an output which looks like:
```

![Supercell of NaCl density](NaCl_den_2x2.png){width="50%"}

> This is what we'd expect of a the charge density in an ionic crystal: a regular pattern where dot with a low relative electron density (the cation) is next to one with a high electron density (the anion).

Reviewer: "where dot with"? This doesn't make sense to me.

Now let's follow the exact same procedure, except this time with `NaCl_elf` for the ELF result. This is the outcome:

![Supercell of NaCl ELF](NaCl_elf_2x2.png){width="50%"}

> This is rather interesting: though the electron density around the Na^+^ cation is very low, the electrons are much more localised around it - the opposite is true for Cl^-^ : the results have virtually inversed. While both, in the simplest model, have full outer shells and thus should both be highly localised (in fact they are both localised, Na^+^ just moreso), Cl^-^ simply has larger orbitals and has a full valence shell (rather than core), leading to lower localisation. This helps illustrate the key difference between ELF and population analysis: ELF is purely a measure of how localised the electrons are in space, and has little to do with their density

Reviewer: Not sure what is meant be "the results have virtually inversed". Do you mean they're inverted with respect to what we might expect?

Author: I meant that the a region of low value became high value and vice versa - the point I'm trying to make here is that high localisation doesn't always equal high density and low localisation doesn't mean low density.  

I try to emphasise how ELF is different from charge density, since that confused me at first

## Covalent bonding - Si

We will now use ELF to demonstrate covalent bonding in diamond-silicon. We will use the `cell` file

*Si.cell*

```

%block lattice_abc
3.8 3.8 3.8
60 60 60
%endblock lattice_abc

%block positions_frac
Si 0.00 0.00 0.00
Si 0.25 0.25 0.25
%endblock positions_frac
```

> Run castep with an identical `param` file (you may remove `WRITE_FORMATTED_DENSITY : TRUE` if not you're not interested in the charge density). Do the exact same procedure as above (though again you can ignore anything to do with the charge density). After opening it in Vesta, rather than taking a slice, click `Edit -> Lattice planes`, which will open the window

Reviewer: ```suggestion
Run castep with an identical `param` file (you may remove `WRITE_FORMATTED_DENSITY : TRUE` if not you're not interested in the charge density). Follow the exact same procedure as above (though again you can ignore anything to do with the charge density). After opening it in Vesta, rather than taking a slice, click `Edit -> Lattice planes`, which will open the window below:
```

![Lattice planes window](lattice_planes_window.png){width="30%"}

> Click `New` and use the (100) plane as before. Then click `Ok`, and again create a supercell - this time click `Boundary...` on the left side of the main window. Make it a 2x2 supercell as before. Rotate the cell around until you see what's happening well - you should see something like

Reviewer: For consistency with above.
```suggestion
1. Click `New` and use the `100` plane as before. 
2. Then click `Ok`.
3. Create a supercell.
4. This time click `Boundary...` on the left side of the main window. 
5. Make a 2x2 supercell as before. 
6. Rotate the cell around until you see what's happening.

 You should see something like:
```

![100 lattice ELF Si](Si_ELF.png){width="50%"}

You can uncheck `Show models` to see it better without the atoms

![100 lattice ELF Si no atoms](Si_ELF2.png){width="40%"}

This tells us a rather interesting story - the atoms are highly localised along the covalent Si-Si bonds, but relatively delocalised near the atoms themselves. They are especially delocalised in regions where is no bonding or atoms.

> If you wish to also examine and compare the charge density, this is covered in [a previous tutorial](charge_density.md)

Reviewer: "previous" implies there's some required order to these, whereas I think they're more down to needs. 
```suggestion
If you wish to also examine and compare the charge density, this is covered in [another tutorial](charge_density.md)
```

## HF

> Now that we have a good understand of what ELF actually does, let's use it for a more interesting purpose: to demonstrate lone pairs and hydrogen bonding. Lone pairs are by their nature highly localised, pairing up with each other and being held largely in place by strong repulsions to any negative charges (including other lone pairs) and strong attractions to positive ones (such as adjacent dipoles - this leads to hydrogen bonding)

Reviewer: > pairing up with each other

As far as I understand, lone pairs don't pair up with each other as is implied here then subsequently contradicted.

> strong repulsions to any negative charges (including other lone pairs)

Not sure what that's trying to say.

Author: Don't they pair up with each other because of the Pauli exclusion principle? And the point I'm trying to make with their repulsions is that they are fixed in a specific position due to the balance of repulsion + attraction, leading to their high localisation and thus why ELF is good for seeing them (at least this is how I understand it - I might just be wrong)

For HF we will use the `cell` file:

*HF.cell*
```
%block lattice_abc
6 6 6
90 90 90
%endblock lattice_abc

%block positions_abs
H 3 3 3
F 3.92 3 3
%endblock positions_abs

kpoints_mp_grid 4 4 4

```

> and again the same `param` file. Following the exact same steps as with NaCl, slicing the (010) plane (with the distance from the origin, set in the same slice window, set as 0.5d) we get an output looking like:

Reviewer: ```suggestion
and again the same `param` file. Following the exact same steps as with NaCl, slicing the `010` plane (with the distance from the origin, set in the same slice window, set as 0.5d) we get an output looking like:
```
What is "0.5d"?

![NH3 result](NH3_slice.png){width=30%}

Here we see evidence of lone pairs on both of the hydrogens in the plane (as both can form hydrogen bonds - and so can the third if you look at that one). This is rather unexpected - in the classical view, NH~3~ has a nitrogen with 1 lone pair that simply forms a single hydrogen bond with a hydrogen in an adjacent NH~3~. However, because every hydrogen is capable of forming a hydrogen bond, ELF registers them as having regions of high locality near them. The same applies to nitrogen - though it should clasically only have 1 pair near it, it is seen as having 2 as it is capable of making 2 hydrogen bonds in that plane.

!!! warning

>     The results will likely look different depending on which 2 hydrogens you pick along with the nitrogen. This can be mostly explained due to how adjacent ammonia molecules are aligned relative to it, which is defined by the cell.

Reviewer: Need to mention periodic boundaries here?
